salt/modules/panos.py

```diff
@@ -0,0 +1,1854 @@
+# -*- coding: utf-8 -*-
+"""
+Module to provide Palo Alto compatibility to Salt.
+
+:codeauthor: :email:`Spencer Ervin <spencer_ervin@hotmail.com>`
+:maturity:   new
+:depends:    none
+:platform:   unix
+
+
+Configuration
+=============
+This module accepts connection configuration details either as
+parameters, or as configuration settings in pillar as a Salt proxy.
+Options passed into opts will be ignored if options are passed into pillar.
+
+.. seealso::
+    :prox:`Palo Alto Proxy Module <salt.proxy.panos>`
+
+About
+=====
+This execution module was designed to handle connections to a Palo Alto based
+firewall. This module adds support to send connections directly to the device
+through the XML API or through a brokered connection to Panorama.
+
+"""
+
+# Import Python Libs
+from __future__ import absolute_import
+import logging
+import time
+
+# Import Salt Libs
+import salt.proxy.panos
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'panos'
+
+
+def __virtual__():
+    """
+    Will load for the panos proxy minions.
+    """
+    try:
+        if salt.utils.is_proxy() and \
+           __opts__['proxy']['proxytype'] == 'panos':
+            return __virtualname__
+    except KeyError:
+        pass
+
+    return False, 'The panos execution module can only be loaded for panos proxy minions.'
+
+
+def _get_job_results(query=None):
+    """
+    Executes a query that requires a job for completion. This funciton will wait for the job to complete
+    and return the results.
+    """
+    if query is None:
```

Comment:
Other than personal preference, I wanted to account for the odd scenario where a variable I was checking for existence was 0. The comparative with 'is None' should ensure that the value has not been declared, while the 'not' statement will potentially catch cases that are not desired.

0 is None: returns false, as it is declared
not 0: returns True, and is often not the desired out come.

I have updated to utilize the 'if not' check. I don't see any current uses that this would cause concern.